### PR TITLE
refactor: optimize l1 image

### LINKED ISSUE #63: Middle: optimizing the taiko-l1 image
Before this optimize: 9 minutes
After optimizing the taiko-l1 image: 5 minutes

### PR DESCRIPTION
**Pull request by alexshliu**

### CODE REVIEW

File: taiko/roles.go [hunk 1]
```diff
@@ -75,8 +77,10 @@ func NewProverConfig(env *TestEnv) *prover.Config {
 	}
 }
 
-func NewProver(t *hivesim.T, env *TestEnv, c *prover.Config) *prover.Prover {
+func NewProver(t *hivesim.T, env *TestEnv, c *prover.Config) (*prover.Prover, error) {
 	p := new(prover.Prover)
-	prover.InitFromConfig(env.Context, p, c)
-	return p
+	if err := prover.InitFromConfig(env.Context, p, c); err != nil {
```
Comment (johntaiko) [inline taiko/roles.go:82]:
```suggestion
	return prover.InitFromConfig(env.Context, p, c)
```

File: taiko/roles.go [hunk 2]
```diff
@@ -63,10 +63,12 @@ func NewProposerConfig(env *TestEnv, l1, l2 *ELNode) *proposer.Config {
 	}
 }
 
-func NewProposer(t *hivesim.T, env *TestEnv, c *proposer.Config) *proposer.Proposer {
+func NewProposer(t *hivesim.T, env *TestEnv, c *proposer.Config) (*proposer.Proposer, error) {
 	p := new(proposer.Proposer)
-	proposer.InitFromConfig(env.Context, p, c)
-	return p
+	if err := proposer.InitFromConfig(env.Context, p, c); err != nil {
```
Comment (johntaiko) [inline taiko/roles.go:68]:
```suggestion
	return proposer.InitFromConfig(env.Context, p, c)
```

Comment (alexshliu) [reply taiko/roles.go:68]: proposer.InitFromConfig just return an error, but NewProposer return two value

File: taiko/roles.go (hunk 1 above)
Comment (alexshliu) [reply taiko/roles.go:82]: same reason as line 68

**Review by johntaiko — APPROVED** [review]

**Pull request merged**